lib/init/grass.py

```diff
@@ -1284,11 +1249,7 @@ def load_gisrc(gisrc, gisrcrc):
 
     :returns: MapsetSettings object
     """
-    mapset_settings = MapsetSettings()
-    kv = read_gisrc(gisrc)
-    mapset_settings.gisdbase = kv.get("GISDBASE")
-    mapset_settings.location = kv.get("LOCATION_NAME")
-    mapset_settings.mapset = kv.get("MAPSET")
+    mapset_settings = getMapsetSettings(gisrc)
     if not mapset_settings.is_valid():
```

Comment:
When it is not valid, your function returns None, not an (invalid) settings, so this like fails. The right solution probably depends on how is_valid is defined.